bundles/org.openhab.binding.speedtest/src/main/java/org/openhab/binding/speedtest/internal/SpeedtestHandler.java

```diff
@@ -0,0 +1,536 @@
+        String ch = channelUID.getId();
+        if (command instanceof RefreshType) {
+            if (!server.isBlank()) {
+                updateChannels();
+            }
+            return;
+        }
+        if (ch.equals(SpeedtestBindingConstants.TRIGGER_TEST)) {
+            if (command instanceof OnOffType) {
+                if (command == OnOffType.ON) {
+                    getSpeed();
+                    updateState(channelUID, OnOffType.OFF);
+                }
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(SpeedtestConfiguration.class);
+        pollingInterval = config.refreshInterval;
+        serverID = config.serverID;
+        if (!config.execPath.isEmpty()) {
+            speedTestCommand = config.execPath;
+        } else {
+            switch (getOperatingSystemType()) {
+                case WINDOWS:
+                    speedTestCommand = "";
+                    break;
+                case LINUX:
+                case MAC:
+                case SOLARIS:
+                    speedTestCommand = "/usr/bin/speedtest";
+                    break;
+                default:
+                    speedTestCommand = "";
+            }
+        }
+
+        updateStatus(ThingStatus.UNKNOWN);
+
+        if (!checkConfig(speedTestCommand)) { // check the config
+            return;
+        }
+        if (!getSpeedTestVersion()) {
+            return;
+        }
+        getServerList();
+        updateStatus(ThingStatus.ONLINE);
+        isRunning = true;
+        onUpdate(); // Setup the scheduler
+    }
+
+    /**
+     * This is called to start the refresh job and also to reset that refresh job when a config change is done.
+     */
+    private void onUpdate() {
+        logger.debug("Polling Interval Set: {} ", pollingInterval);
+        if (pollingInterval > 0) {
+            if (pollingJob == null || pollingJob.isCancelled()) {
+                pollingJob = scheduler.scheduleWithFixedDelay(pollingRunnable, 0, pollingInterval, TimeUnit.MINUTES);
+            }
+        }
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Disposing Speedtest Handler Thing");
+        isRunning = false;
+        if (pollingJob != null && !pollingJob.isCancelled()) {
+            pollingJob.cancel(true);
+            pollingJob = null;
+        }
+    }
+
+    /**
+     * Called when this thing gets it's configuration changed.
+     */
+    @Override
+    public void thingUpdated(Thing thing) {
+        dispose();
+        this.thing = thing;
+        initialize();
+    }
+
+    /**
+     * Polling event used to get speed data from speedtest
+     */
+    private Runnable pollingRunnable = () -> {
+        try {
+            getSpeed();
+        } catch (Exception e) {
+            logger.warn("An exception occurred while running Speedtest: '{}'", e.getMessage());
+            updateStatus(ThingStatus.OFFLINE);
+        }
+    };
+
+    /*
```

Comment:
All method JavaDocs should should start with `**`

```suggestion
    /**
```